.devops/azure-templates/setup-project.yml

```diff
@@ -0,0 +1,34 @@
+# Azure DevOps pipeline template used to setup the Node project:
+# 1. checkout code
+# 2. setup correct node version
+# 3. install node dependencies
+
+parameters:
+- name: 'cache_version_id'
+  type: string
+  default: $(CACHE_VERSION_ID)
+
+# the branch, tag or commit to deploy
+- name: 'gitReference'
+  type: string
+  default: '$(Build.SourceVersion)'
+  
+steps:
+- checkout: self
+  displayName: 'Checkout'
+
+# This is needed because the pipeline may point to a different commit than expected
+# The common case is when the previous stage pushed another commit
+- ${{ if ne(parameters.gitReference, variables['Build.SourceVersion']) }}:
+  - script: |
+      git fetch && git checkout ${{ parameters.gitReference }}
+    displayName: 'Checkout reference'  
+
+- task: UseNode@1
+  inputs:
+    version: $(NODE_VERSION)
+  displayName: 'Set up Node.js'
+
+- script: |
+    yarn install --frozen-lockfile --no-progress --non-interactive --network-concurrency 1
+  displayName: 'Install node modules'
```

Comment:
Since this template is shared with other projects (like io-functions-app) we can think about moving it into azure-pipeline-tempates repo in future

Reply:
I agree. you can propose a PR at https://github.com/pagopa/azure-pipeline-templates